cpp/benchmarks/io/text/multibyte_split.cpp

```diff
@@ -103,22 +103,23 @@ static void bench_multibyte_split(nvbench::state& state)
   auto const delim_factor = static_cast<double>(delim_percent) / 100;
   auto device_input       = create_random_input(file_size_approx, delim_factor, 0.05, delim);
   auto host_input         = thrust::host_vector<char>(device_input.size());
-  auto const host_string  = std::string(host_input.data(), host_input.size());
 
   cudaMemcpyAsync(host_input.data(),
                   device_input.data(),
                   device_input.size() * sizeof(char),
                   cudaMemcpyDeviceToHost,
                   cudf::default_stream_value);
 
+  cudaDeviceSynchronize();
+
   auto const temp_file_name = random_file_in_dir(temp_dir.path());
 
   {
     auto temp_fostream = std::ofstream(temp_file_name, std::ofstream::out);
     temp_fostream.write(host_input.data(), host_input.size());
   }
 
-  cudaDeviceSynchronize();
+  auto const host_string = std::string(host_input.data(), host_input.size());
```

Comment:
why do we need a second copy of the input data? Can we use the string directly instead of the `thrust::host_vector` object?

Reply:
great point, I now only allocate and copy the host data if necessary.